openbas-api/src/main/java/io/openbas/executors/crowdstrike/service/CrowdStrikeExecutorContextService.java

```diff
@@ -55,48 +57,178 @@ public void launchExecutorSubprocess(
             .orElseThrow(
                 () -> new UnsupportedOperationException("Inject does not have a contract"));
 
-    String scriptName;
-    String implantLocation;
-    switch (platform) {
-      case Windows -> {
-        scriptName = this.crowdStrikeExecutorConfig.getWindowsScriptName();
-        implantLocation =
-            "$location="
-                + IMPLANT_LOCATION_WINDOWS
-                + ExecutorHelper.IMPLANT_BASE_NAME
-                + UUID.randomUUID()
-                + "\";md $location -ea 0;[Environment]::CurrentDirectory";
-      }
-      case Linux, MacOS -> {
-        scriptName = this.crowdStrikeExecutorConfig.getUnixScriptName();
-        implantLocation =
-            "location="
-                + IMPLANT_LOCATION_UNIX
-                + ExecutorHelper.IMPLANT_BASE_NAME
-                + UUID.randomUUID()
-                + ";mkdir -p $location;filename=";
+    manageWithoutPlatformAgents(agents, injectStatus);
+    List<CrowdStrikeAction> actions = new ArrayList<>();
+    // Set implant script for Windows CS agents
+    actions.addAll(
+        getWindowsActions(
+            agents.stream()
```

Comment:
Can be extracted to it's own method for better readability (private List<CrowdStrikeAction> getActionsForOS(List<Agent> agents, Endpoint.PLATFORM_TYPE)